`-Wduplicated-branches` warning in `wayland_embedder.cpp`

### Tested versions

reproducible on master

build with "scons debug_symbols=yes dev_mode=yes tests=yes" on ubuntu 24.04.3 LTS

### System information

ubuntu 24.04.3 LTS

### Issue description

build broken: platform/linuxbsd/wayland/wayland_embedder.cpp:1118:41: error: this condition has identical branches [-Werror=duidentical branches [-Werror=duplicated-branches]plicated-branches]

### Steps to reproduce

reproducible on master

build with "scons debug_symbols=yes dev_mode=yes tests=yes" on ubuntu 24.04.3 LTS

### Minimal reproduction project (MRP)

n/a

--- a/platform/linuxbsd/wayland/wayland_embedder.cpp
+++ b/platform/linuxbsd/wayland/wayland_embedder.cpp
@@ -1115,11 +1115,13 @@ bool WaylandEmbedder::handle_generic_msg(Client *client, const WaylandObject *p_
 				}
 
 				if (new_interface == nullptr) {
+#ifdef WAYLAND_EMBED_DEBUG_LOGS_ENABLED
 					if (last_str_len > 0) {
 						DEBUG_LOG_WAYLAND_EMBED(vformat("Unknown interface %s, marking packet as invalid.", (char *)(body + last_str_buf_idx + 1)));
 					} else {
 						DEBUG_LOG_WAYLAND_EMBED("Unknown interface, marking packet as invalid.");
 					}
+#endif
 					valid = false;
 					break;
 				}

